core/src/main/java/org/apache/iceberg/ReplaceManifests.java

```diff
@@ -0,0 +1,157 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg;
+
+import com.google.common.base.Preconditions;
+import com.google.common.collect.Lists;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.atomic.AtomicInteger;
+import javax.annotation.concurrent.ThreadSafe;
+import org.apache.iceberg.exceptions.RuntimeIOException;
+import org.apache.iceberg.util.Tasks;
+import org.apache.iceberg.util.ThreadPools;
+
+
+@ThreadSafe
+public class ReplaceManifests extends SnapshotProducer<RewriteManifests> implements RewriteManifests {
+  private final SnapshotSummary.Builder summaryBuilder = SnapshotSummary.builder();
+  private final List<ManifestFile> newManifests = Collections.synchronizedList(Lists.newArrayList());
+  private final List<ManifestFile> existingManifests = Collections.synchronizedList(Lists.newArrayList());
+  private final PartitionSpec spec;
+  private final Map<Object, ManifestWriter> writers = Collections.synchronizedMap(new HashMap<>());
+  private AtomicInteger manifestCount = new AtomicInteger(0);
+
+  ReplaceManifests(TableOperations ops) {
+    super(ops);
+    this.spec = ops.current().spec();
+  }
+
+  @Override
+  protected String operation() {
+    return DataOperations.APPEND;
```

Comment:
This isn't an append because no data is added. Instead, use `DataOperations.REPLACE` to signal that the data in the table did not change.

Reply:
Sure, I was going back and forth which to use. I'll change it...